src/x/time/unit.go

```diff
@@ -153,28 +154,34 @@ func MaxUnitForDuration(d time.Duration) (int64, Unit) {
 }
 
 var (
-	unitStrings = map[Unit]string{
-		Second:      "s",
-		Millisecond: "ms",
-		Nanosecond:  "ns",
-		Microsecond: "us",
-		Minute:      "m",
-		Hour:        "h",
-		Day:         "d",
-		Year:        "y",
+	unitStrings = []string{
+		"unknown",
+		"s",
+		"ms",
+		"us",
+		"ns",
+		"m",
+		"h",
+		"d",
+		"y",
 	}
 
-	durationsToUnit = make(map[time.Duration]Unit)
-	unitsToDuration = map[Unit]time.Duration{
+	// Using an array here to avoid map access cost.
+	unitsToDuration = []time.Duration{
+		None:        time.Duration(0),
 		Second:      time.Second,
 		Millisecond: time.Millisecond,
-		Nanosecond:  time.Nanosecond,
 		Microsecond: time.Microsecond,
+		Nanosecond:  time.Nanosecond,
 		Minute:      time.Minute,
 		Hour:        time.Hour,
 		Day:         time.Hour * 24,
 		Year:        time.Hour * 24 * 365,
 	}
+	durationsToUnit = make(map[time.Duration]Unit)
+
+	unitCount = len(unitsToDuration)
```

Comment:
Maybe we should export this here and then set 
`encodingSchemes := make(TimeEncodingSchemes, xtime.UnitCount)` in `schemes.go:NewTimeEncodingSchemes`

Reply:
Yeah good idea